src/Database/Schema/Migrations/Filesystem.hs

```diff
@@ -106,79 +101,77 @@ migrationFromPath path = do
-
-requiredFields :: [Text]
-requiredFields = [ "Apply"
-                 , "Depends"
-                 ]
-
-fieldProcessors :: [(Text, FieldProcessor)]
-fieldProcessors = [ ("Created", setTimestamp )
-                  , ("Description", setDescription )
-                  , ("Apply", setApply )
-                  , ("Revert", setRevert )
-                  , ("Depends", setDepends )
-                  ]
-
-setTimestamp :: FieldProcessor
-setTimestamp value m = do
-  ts <- case readTimestamp value of
-          [(t, _)] -> return t
-          _ -> fail "expected one valid parse"
-  return $ m { mTimestamp = Just ts }
-
-readTimestamp :: Text -> [(UTCTime, String)]
-readTimestamp = reads . cs
-
-setDescription :: FieldProcessor
-setDescription desc m = Just $ m { mDesc = Just desc }
-
-setApply :: FieldProcessor
-setApply apply m = Just $ m { mApply = apply }
-
-setRevert :: FieldProcessor
-setRevert revert m = Just $ m { mRevert = Just revert }
-
-setDepends :: FieldProcessor
-setDepends depString m = Just $ m { mDeps = T.words depString }
+        then Yaml.decodeFileThrow (addNewMigrationExtension path) `catch` (\(e::Yaml.ParseException) -> throwFS $ show e)
+        else Yaml.decodeFileThrow (addMigrationExtension path filenameExtensionTxt) `catch` (\(e::Yaml.ParseException) -> throwFS $ show e)
+
+    process name = migrationYamlToMigration name <$> readMigrationFile
+
+-- | TODO: re-use this for the generation side too
+data MigrationYaml = MigrationYaml
+    { myCreated :: Maybe UTCTimeYaml
+    , myDescription :: Maybe Text
+    , myApply :: Text
+    , myRevert :: Maybe Text
+    , myDepends :: DependsYaml
+    }
+    deriving Generic
+
+instance FromJSON MigrationYaml where
+    parseJSON = genericParseJSON jsonOptions
+
+instance ToJSON MigrationYaml where
+    toJSON = genericToJSON jsonOptions
+    toEncoding = genericToEncoding jsonOptions
+
+jsonOptions :: Options
+jsonOptions = defaultOptions
+    { fieldLabelModifier = drop 2 -- remove "my" prefix
+    , omitNothingFields = True
+    , rejectUnknownFields = True
+    }
+
+migrationYamlToMigration :: Text -> MigrationYaml -> Migration
+migrationYamlToMigration theId my = Migration
+    { mTimestamp = unUTCTimeYaml <$> myCreated my
+    , mId = theId
+    , mDesc = myDescription my
+    , mApply = myApply my
+    , mRevert = myRevert my
+    , mDeps = unDependsYaml $ myDepends my
+    }
+
+newtype UTCTimeYaml = UTCTimeYaml
+    { unUTCTimeYaml :: UTCTime
+    }
+
+instance FromJSON UTCTimeYaml where
+    parseJSON = withText "UTCTime"
+        $ maybe (fail "Unable to parse UTCTime") (pure . UTCTimeYaml)
+        . parseTimeM True defaultTimeLocale utcTimeYamlFormat
+        . cs
+
+instance ToJSON UTCTimeYaml where
+    toJSON = toJSON . formatTime defaultTimeLocale utcTimeYamlFormat . unUTCTimeYaml
+    toEncoding = toEncoding . formatTime defaultTimeLocale utcTimeYamlFormat . unUTCTimeYaml
+
+-- Keeps things as the old Show/Read-based format, e.g "2009-04-15 10:02:06 UTC"
```

Comment:
:wave: @jtdaugherty I just wanted to mention: I finally updated my own app to this version of this library and I found it failed because my migrations all had fractional seconds, such as

```yaml
Created: 2009-04-15 10:02:06.23345 UTC
```

I'm not sure where that came from, or why the parse won't accept it, because I _swear_ I tested things when I wrote this PR.

In any event, I ran the following to truncate them before my migrations would run with the new version:

```console
sed -i 's/\.[0-9]\+ UTC$/ UTC/' db/migrations/**/*  
```

I'm not sure if we should make an update to make the parser more flexible?

Reply:
Yikes! Sorry to hear about this. It seems to me that a relaxation to the parser ought to be okay. If you want to make that change, let me know - otherwise I'll see to it.

Reply:
> It seems to me that a relaxation to the parser ought to be okay

Yeah, agreed. If I'm reading the docs correctly, it seems like using `...%T%Q UTC` could do it:

> **%Q**
>
> decimal point and fraction of second, up to 12 second decimals, without trailing zeros. **For a whole number of seconds, %Q omits the decimal point** unless padding is specified

(Emphasis mine)

I'm not quite sure my availability, so I'll comment again if I get a chance to work on it; otherwise, feel free.

Reply:
I think we need to do something, because new migrations still get created with fractional seconds, so the problem will keep coming back. @pbrisbin any chance you could create a PR with this fix, as you seem quite familiar with the problem and relevant code?

Reply:
> because new migrations still get created with fractional seconds

I didn't realize this. You're saying this version _creates_ fractional seconds files that it can't parse? That's much worse.

> you seem quite familiar with the problem and relevant code?

I may have been at one time, but that was a long time ago. Still, it seems like a relatively straightforward thing. The decoder needs to accept with or without fractional seconds. Adding a unit test and using the format-string I mentioned above should do it.